### PR TITLE
Add ZNPCsPlus, ServerNPC and PinataParty support

### PR DESCRIPTION
**Pull request by Krakenied**
Closes https://github.com/LMBishop/Quests/issues/593
Closes https://github.com/LMBishop/Quests/issues/518

### CODE REVIEW

**Comment by Krakenied** [comment]
https://github.com/LMBishop/Quests/issues/593: @WarnDa, @NoltoxGit

I think it works, the build can be downloaded here: https://github.com/LMBishop/Quests/actions/runs/7914189686
Could you test it and confirm it can be merged?

**Comment by WarnDa** [comment]
I just tried to set quests to "znpcsplus_deliver" and the good ID, but it does nothing when I interact with the npc
And on /q a reload I got no errors relating to invalid quest.

image
image

**Comment by Krakenied** [comment]
This id should be used
image

**Comment by Krakenied** [comment]
Ok I see, you should use `'2'` instead of `2` - Quests see it as number, not a string

**Comment by Krakenied** [comment]
Ok tested it and yeah, it doesn't work but don't know why yet

**Comment by Krakenied** [comment]
wait, it works, forgot to start the quest XD

File: bukkit/libs/ServerNPC-1.16.1-b1.jar
Comment (Krakenied) [inline bukkit/libs/ServerNPC-1.16.1-b1.jar:1]: [ServerNPC](https://www.spigotmc.org/resources/servernpc.74524/) by @iSnakeBuzz

This JAR contains only API classes necessary for the task type implementation

**Comment by NoltoxGit** [comment]
> #593: @WarnDa, @NoltoxGit
> 
> I think it works, the build can be downloaded here: https://github.com/Krakenied/Quests/actions/runs/7914189418/artifacts/1247499746 Could you test it and confirm it can be merged?

Hello,
I have the impression that the task `znpcsplus_interact` is not working, is this normal ?

**Comment by NoltoxGit** [comment]
> > #593: @WarnDa, @NoltoxGit
> > I think it works, the build can be downloaded here: https://github.com/Krakenied/Quests/actions/runs/7914189418/artifacts/1247499746 Could you test it and confirm it can be merged?
> 
> Hello, I have the impression that the task `znpcsplus_interact` is not working, is this normal ?

And I have a lot of errors in my server : https://mclo.gs/Z8V1JP2#L1388

**Comment by Krakenied** [comment]
Going to test with Citizens and ServerNPC and will mark as ready for review

**Comment by Krakenied** [comment]
> > Going to test with Citizens and ServerNPC and will mark as ready for review
> 
> I will test with the new action `3.14.2-afe1c44` and the task deliver works perfectly. But not for the `znpcsplus_interact` task.

Just implemented it, try with this one https://github.com/Krakenied/Quests/actions/runs/7916550779

**Comment by Krakenied** [comment]
Citizens and ZNPCsPlus seems to work properly, waiting for feedback from my friend who'll test it with ServerNPC

**Comment by NoltoxGit** [comment]
> > > > #593: @WarnDa, @NoltoxGit
> > > > I think it works, the build can be downloaded here: https://github.com/Krakenied/Quests/actions/runs/7914189418/artifacts/1247499746 Could you test it and confirm it can be merged?
> > > 
> > > 
> > > Hello, I have the impression that the task `znpcsplus_interact` is not working, is this normal ?
> > 
> > 
> > And I have a lot of errors in my server : https://mclo.gs/Z8V1JP2#L1388
> 
> Already fixed.
> 
> > > Going to test with Citizens and ServerNPC and will mark as ready for review
> > 
> > 
> > I will test with the new action `3.14.2-afe1c44` and the task deliver works perfectly. But not for the `znpcsplus_interact` task.
> 
> Just implemented it, try with this one https://github.com/Krakenied/Quests/actions/runs/7916550779

That sounds all good to me, thank you very much ! ❤️

**Comment by Krakenied** [comment]
https://github.com/LMBishop/Quests/issues/518: @NoltoxGit

I think it works, the build can be downloaded here: https://github.com/LMBishop/Quests/actions/runs/7927494188/artifacts/1250482979
Could you test it and confirm it works and can be merged?

Example task:
```
pinata:
  type: "pinataparty_hit"
  amount: 100 # amount of hits
  worlds:
  - "world"

File: bukkit/libs/PinataParty-2.63.9-API.jar
Comment (Krakenied) [inline bukkit/libs/PinataParty-2.63.9-API.jar:1]: [PinataParty](https://www.spigotmc.org/resources/pinataparty-interactive-event-%E2%9C%A8-aio-voting-system-%E2%9C%A8-1-12-x-1-20-x.59318/) by @HexedHero

This JAR contains only API classes necessary for the task type implementation

**Comment by NoltoxGit** [comment]
> #518: @NoltoxGit
> 
> I think it works, the build can be downloaded here: https://github.com/LMBishop/Quests/actions/runs/7927494188/artifacts/1250482979 Could you test it and confirm it works and can be merged?
> 
> Example task:
> 
> ```
> pinata:
>   type: "pinataparty_hit"
>   amount: 100 # amount of hits
>   worlds:
>   - "world"
> ```

Ohhhh, i will try this, thank you so much :D

**Comment by NoltoxGit** [comment]
> amount

It seems to work well, thank you very much :)

**Comment by LMBishop** [comment]
> Citizens and ZNPCsPlus seems to work properly, waiting for feedback from my friend who'll test it with ServerNPC

Let me know when this is ready for merge :)

I will update the documentation for 3.15 in the meantime.

**Comment by Krakenied** [comment]
It's ready tbh

**Review by LMBishop — APPROVED** [review]

**Pull request merged**